jax/scipy/sparse/linalg.py

```diff
@@ -23,10 +23,23 @@
 from jax.tree_util import tree_leaves, tree_map, tree_multimap
 
 
+def _vdot_real_part(x, y):
+  """Vector dot-product guaranteed to have a real valued result."""
+  # all our uses of vdot() in CG are for computing an operator of the form
+  # `z^T M z` where `M` is positive definite, and hence Hermitian, so the
```

Comment:
Some authors, like Golub and Van Loan, take "positive definite" not to imply symmetric (but rather just positive real eigenvalues), but I have a feeling you already knew that :P Sorry, couldn't resist nitting. Don't feel obligated to change it.

Reply:
Good point! For `scipy.linalg.cg`, the requirement is already that the matrix is Hermitian and positive definite, so I'll just note that instead